app/modules/cart/repository.py

```diff
@@ -1,22 +1,95 @@
+from sqlalchemy import select
+from sqlalchemy.exc import SQLAlchemyError
 from sqlalchemy.ext.asyncio import AsyncSession
 
+from app.common.exceptions.app_exceptions import DatabaseOperationException, EntityNotFoundException
+
 from .models import CartItem
 from .repository_interface import CartItemRepositoryInterface
 
 
 class CartItemRepository(CartItemRepositoryInterface):
+    model_class = CartItem
+
     def __init__(self, session: AsyncSession):
         self.session = session
 
     async def insert(self, cart_item: CartItem) -> CartItem:
-        self.session.add(cart_item)
-        await self.session.commit()
-        await self.session.refresh(cart_item)
-        return cart_item
+        try:
+            self.session.add(cart_item)
+            await self.session.commit()
+            await self.session.refresh(cart_item)
+            return cart_item
+        except SQLAlchemyError as e:
+            await self.session.rollback()
+            raise DatabaseOperationException(
```

Comment:
we usually handle DB exceptions at a higher layer (e.g., service layer/usecase) rather than the repository itself, to keep our repositories focused on data access logic and avoid coupling them with application-specific error handling.